src/Build/Evaluation/Conditionals/NotEqualExpressionNode.cs

```diff
@@ -15,17 +15,17 @@ internal sealed class NotEqualExpressionNode : MultipleComparisonNode
         /// <summary>
         /// Compare numbers
         /// </summary>
-        protected override bool Compare(double left, double right)
+        protected override bool Compare(double left, double? right)
         {
-            return left != right;
+            return right is null || left != right.Value;
```

Comment:
nit: I'd check this in the caller to avoid the asymmetry. Should be easy if you switch to the `bool Try(.., out value)` pattern suggested in my other comment.